msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/AccountState.kt

```diff
@@ -227,42 +230,79 @@ class AccountState private constructor(
      * @return [com.microsoft.identity.nativeauth.statemachine.results.GetAccessTokenResult] The result of the getAccessToken action
      */
     suspend fun getAccessToken(forceRefresh: Boolean = false): GetAccessTokenResult {
+        return getAccessTokenInternal(forceRefresh, null);
+    }
+
+    /**
+     * Retrieves the access token for the currently signed in account from the cache.
+     * If the access token is expired, it will be attempted to be refreshed using the refresh token that's stored in the cache;
+     * Kotlin coroutines variant.
+     *
+     * @return [com.microsoft.identity.nativeauth.statemachine.results.GetAccessTokenResult] The result of the getAccessToken action
+     */
+    suspend fun getAccessToken(forceRefresh: Boolean = false, scopes: List<String>): GetAccessTokenResult {
+        var scopeSet = HashSet<String>()
+        if (!scopes.isNullOrEmpty()) {
+            scopeSet.addAll(scopes)
+        }
+        val scopeSet2 = BaseController.addDefaultScopes(scopeSet)
+        return getAccessTokenInternal(forceRefresh, ArrayList(scopeSet2))
+    }
+
+    suspend fun getAccessTokenInternal(forceRefresh: Boolean, scopes: List<String>?): GetAccessTokenResult {
         LogSession.logMethodCall(
             tag = TAG,
             correlationId = null,
             methodName = "$TAG.getAccessToken(forceRefresh: Boolean)"
         )
         return withContext(Dispatchers.IO) {
-            val account =
-                NativeAuthPublicClientApplication.getCurrentAccountInternal(config) as? Account
+
+            val accountResult = NativeAuthPublicClientApplication.getCurrentAccountInternal(config) as? Account
+            val account = accountResult
                     ?: return@withContext GetAccessTokenError(
                         errorType = GetAccessTokenErrorTypes.NO_ACCOUNT_FOUND,
                         error = MsalClientException.NO_CURRENT_ACCOUNT,
                         errorMessage = MsalClientException.NO_CURRENT_ACCOUNT_ERROR_MESSAGE,
                         correlationId = "UNSET"
                     )
 
-            val acquireTokenSilentParameters = AcquireTokenSilentParameters.Builder()
+            val acquireTokenSilentParametersBuilder = AcquireTokenSilentParameters.Builder()
```

Comment:
What is the difference between `AcquireTokenSilentParameters` and `SilentTokenCommandParameters`? Why are we using one and not the other?

Reply:
Ah, `AcquireTokenSilentParameters` is for the "interface commands" and `SilentTokenCommandParameters` is for the "internal commands"? Is there a need for us to use `AcquireTokenSilentParameters` - i.e. can we directly use `SilentTokenCommandParameters`?

Reply:
That would require more code then present. We will have to create a new method in the `CommandParametersAdapter` class that does not require `AcquireTokenSilentParameters` to be passed.